### ISSUE TITLE
In LinUCBVI.ipynb , running it on colab or locally giving extremely different results

### ISSUE THREAD

**Issue by umangja** [sauxpa/neural_exploration#1]
I have try running your  exact LinUCBVI.ipynb  locally and also on colab but i can't reproduce same results. 

To be concrete, In linear transition prabability and linear reward function (case 1), graph between no of episode vs cumulative regret is comming out to be linear and policy differ extremely from optimal policy and also regret explodes.

I have tried to run many times, results remain different from yours every time. 

you can view results at 
[colab Link](https://colab.research.google.com/drive/1iCdap2d-Iie5lFvCfbe4Cwl9qYzPajrE?usp=sharing)

**Comment by sauxpa** [comment]
Hi Umang,

Thanks for your interest in this. As a full disclaimer, this is an old project of mine which I hadn't touched in a while, and I am not affiliated with the authors of the NeuralUCB papers.

I have been doing some clean up, in particular I have added an option to freeze numpy/torch seeds to help reproducibility.

As a general remark, LinUCBVI provably works on linear MDP (transition and rewards), but in practice can be sensitive to the hyperparameters, in particular the "confidence scale" in the exploration bonus (theory suggests an asymptotic scaling for it but finite time regret is different story). I have fixed a bug in the computation of the exploration bonus and rerun the experiments. 

NeuralUCBVI is conjectural (or at least was at the time), based on the analysis of NeuralUCB as a limit kernel method and existing work on kernelized LinUCBVI.

Have fun!
